Stripe/StripeiOSTests/PaymentSheet+APITest.swift

```diff
@@ -450,6 +449,101 @@ class PaymentSheetAPITest: XCTestCase {
+        let firstUpdateExpectation = expectation(description: "First update callback is invoked")
+        let secondUpdateExpectation = expectation(description: "Second update callback is invoked")
+        var flowController: PaymentSheet.FlowController!
+        PaymentSheet.FlowController.create(intentConfig: intentConfig, configuration: configuration) { result in
+            switch result {
+            case .success(let sut):
+                // This update should succeed
+                flowController = sut
+                intentConfig.mode = .setup(currency: nil, setupFutureUsage: .offSession)
+                flowController.update(intentConfiguration: intentConfig) { error in
+                    XCTAssertNil(error)
+                    firstUpdateExpectation.fulfill()
+
+                    // Subsequent updates should also succeed
+                    intentConfig.mode = .setup(currency: nil, setupFutureUsage: .onSession)
+                    flowController.update(intentConfiguration: intentConfig) { error in
+                        XCTAssertNil(error)
+                        secondUpdateExpectation.fulfill()
+                    }
+                }
+
+            case .failure(let error):
+                XCTFail(error.localizedDescription)
+            }
+        }
+
+        waitForExpectations(timeout: 10)
+    }
+
+    func testUpdateAfterFailedUpdate() {
+        var intentConfig = PaymentSheet.IntentConfiguration(mode: .payment(amount: 1000, currency: "USD")) { _, _ in
+            // These tests don't confirm, so this is unused
+        }
+
+        let failedUpdateExpectation = expectation(description: "First update fails")
+        let secondUpdateExpectation = expectation(description: "Second update callback is invoked")
+        var flowController: PaymentSheet.FlowController!
+        PaymentSheet.FlowController.create(intentConfig: intentConfig, configuration: configuration) { result in
+            switch result {
+            case .success(let sut):
+                flowController = sut
+                // ...updating w/ an invalid intent config should fail...
+                intentConfig.mode = .setup(currency: "Invalid currency", setupFutureUsage: .offSession)
+                flowController.update(intentConfiguration: intentConfig) { updateError in
+                    XCTAssertNotNil(updateError)
+                    failedUpdateExpectation.fulfill()
+
+                    // Subsequent updates should succeed
+                    intentConfig.mode = .setup(currency: "USD", setupFutureUsage: .offSession)
+                    flowController.update(intentConfiguration: intentConfig) { error in
+                        XCTAssertNil(error)
+                        secondUpdateExpectation.fulfill()
+                    }
+                }
+            case .failure(let error):
+                XCTFail(error.localizedDescription)
+            }
+        }
+        waitForExpectations(timeout: 10)
+    }
+
```

Comment:
I wish we could test these two cases explicitly, but I'm not sure we have the tools/helpers to do that right now.  We'd need to have a lot of control over the api client to time its responses.  Leaving this comment in case you have ideas.

![image](https://user-images.githubusercontent.com/47796191/227619524-fd981278-82d1-4ae6-8324-d6d9af011d2b.png)
![image](https://user-images.githubusercontent.com/47796191/227619537-e0a4bdd6-a514-4d65-859e-366a31c2cea2.png)